SignalServiceKit/src/Messages/OWSMessageSender.m

```diff
@@ -939,6 +939,25 @@ - (void)sendMessage:(TSOutgoingMessage *)message
         }
     }
 
+    NSString *localNumber = [TSAccountManager localNumber];
+    if ([localNumber isEqualToString:recipient.uniqueId]) {
+        if (deviceMessages.count < 1) {
```

Comment:
nit: This seems like kind of a weird place for this logic - I'd expect it to be in or near `sendSyncTranscriptForMessage`.

i.e. In the case of sending an attachment to ourself we'd get as far as uploading the attachment before bailing out here.

Reply:
`sendSyncTranscriptForMessage` is only used for sending "sync transcripts".  We want this logic to apply to all sync messages.  Search for ` : OWSOutgoingSyncMessage` for a complete list.